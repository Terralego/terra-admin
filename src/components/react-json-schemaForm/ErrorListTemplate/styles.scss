.error-list {
  margin: 0.5rem 1.5rem 1rem 0;
  padding-top: 1rem;
  padding-bottom: 1rem;
  width: auto;
  .bp3-icon {
    top: 1rem !important;
  }
}
