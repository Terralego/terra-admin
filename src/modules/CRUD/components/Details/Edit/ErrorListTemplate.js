import React from 'react';
import PropTypes from 'prop-types';
import { Callout } from '@blueprintjs/core';

const ErrorListTemplate = ({ errorSchema: { __errors, ...fields }, schema: { properties } }) => {
  const errors = Object.keys(fields).reduce((list, item) => {
    const { __errors: errorsItem } = fields[item];
    return errorsItem.length
      ? [...list, { title: properties[item].title, list: errorsItem }]
      : list;
  }, []);

  return (
    <Callout intent="danger">
      <ul>
        {errors.map(({ title, list }) => (
          <li key={title}>
            {title} :
            <ul>
              {list.map(item => <li key={item}>{item}</li>)}
            </ul>
          </li>
        ))}
      </ul>
    </Callout>
  );
};

ErrorListTemplate.propTypes = {
  errorSchema: PropTypes.shape({}).isRequired,
  schema: PropTypes.shape({
    properties: PropTypes.shape({}),
  }).isRequired,
};

export default ErrorListTemplate;
