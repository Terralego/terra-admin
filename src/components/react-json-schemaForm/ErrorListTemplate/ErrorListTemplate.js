import React from 'react';
import PropTypes from 'prop-types';
import { Callout } from '@blueprintjs/core';
import './styles.scss';

const ErrorListTemplate = ({ t }) => (
  <Callout
    className="error-list"
    intent="danger"
    title={t('jsonSchema.panelError.title')}
  />
);

ErrorListTemplate.propTypes = {
  t: PropTypes.func,
};

ErrorListTemplate.defaultProps = {
  t: () => {},
};

export default ErrorListTemplate;
