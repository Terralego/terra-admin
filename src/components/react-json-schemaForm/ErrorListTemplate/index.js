import { withTranslation } from 'react-i18next';
import ErrorListTemplate from './ErrorListTemplate';

export default withTranslation()(ErrorListTemplate);
