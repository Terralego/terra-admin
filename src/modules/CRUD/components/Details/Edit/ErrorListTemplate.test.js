import React from 'react';
import renderer from 'react-test-renderer';

import ErrorListTemplate from './ErrorListTemplate';

jest.mock('@blueprintjs/core', () => ({
  Callout: ({ children }) => <div>{children}</div>,
}));


const props = {
  errors: [
    { stack: 'geometryFromMap: Il est obligatoire de positionner un objet sur la carte' },
  ],
  errorSchema: {
    description: {
      __errors: [],
    },
    geometryFromMap: {
      __errors: [
        'Il est obligatoire de positionner un objet sur la carte',
      ],
    },
  },
  schema: {
    type: 'object',
    properties: {
      description: {
        type: 'string',
        title: 'Description chapeau',
      },
      geometryFromMap: {
        type: 'boolean',
        title: 'Géométrie',
        default: true,
      },
    },
  },
  uiSchema: {
    description: {
      'ui:widget': 'textarea',
    },
    geometryFromMap: {
      'ui:widget': 'hidden',
    },
  },
};


it('should render correctly', () => {
  const tree = renderer.create((
    <ErrorListTemplate
      {...props}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
