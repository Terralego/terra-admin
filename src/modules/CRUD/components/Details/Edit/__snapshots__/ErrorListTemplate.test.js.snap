// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <ul>
    <li>
      Géométrie
       :
      <ul>
        <li>
          Il est obligatoire de positionner un objet sur la carte
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
